lib/Doctrine/DBAL/Connection.php

```diff
@@ -931,9 +931,10 @@ public function executeQuery($query, array $params = [], $types = [], ?QueryCach
      * @param int[]|string[]    $types  The types the previous parameters are in.
      * @param QueryCacheProfile $qcp    The query cache profile.
      *
-     * @return ResultStatement
+     * @return ResultCacheStatement|ArrayStatement
```

Comment:
Please revert. The type hint should refer to an interface, not to a specific implementation.